server/data/data-cassandra/src/main/java/org/apache/james/user/cassandra/CassandraUsersDAO.java

```diff
@@ -185,7 +183,7 @@ public int countUsers() {
 
     @Override
     public void addUser(Username username, String password) throws UsersRepositoryException {
-        DefaultUser user = new DefaultUser(username, algorithmFactory.of(DEFAULT_ALGO_VALUE));
+        DefaultUser user = new DefaultUser(username, Algorithm.of(DEFAULT_ALGO_VALUE));
```

Comment:
Should we move `Algorithm.of(DEFAULT_ALGO_VALUE)` to outside and make it is a static final variable?

Reply:
Please check the second commit, there it is in member variables `preferredAlgorithm` and `fallbackAlgorithm`, which are initialized from configuration file.